Load balance travis tasks to reduce wall clock time.

crankyoldgit opened the pull request:
## Existing run times:
```
Compile the trivial examples
 16 min 6 sec

 Compile the complex examples
 11 min 39 sec

 Unit tests, Linter, & Documentation checks
 4 min 5 sec
```
Longest task: 16+ mins.

## Proposed:
```
 Compile the trivial examples
 9 min 51 sec

 Compile the IRrecvDumpV2+ examples
 10 min 8 sec

 Unit tests, Linter, Doc checks, & Compile IRMQTTServer
 7 min 21 sec
```
Longest task: 10+ mins

crankyoldgit committed:
Load balance the travis compile tasks a bit better.

Currently:
```
Compile the trivial examples
 16 min 6 sec

 Compile the complex examples
 11 min 39 sec

 Unit tests, Linter, & Documentation checks
 4 min 5 sec
```

i.e. move some "complex-ish" builds to the complex task. e.g. IRrecvDumpV3

* Also de-dup the REGEX for determining complex examples.

Fingers crossed, this will reduce the overall (wall) time for travis checks to complete.

crankyoldgit committed:
Oops. That failed. Lets try another approach.

crankyoldgit committed:
Another loadbalancing split.

crankyoldgit committed:
Rename last task.

crankyoldgit committed:
Use correct RE for IRMQTTServer

NiKiZe reviewed (approved):
maybe for make consider: https://docs.travis-ci.com/user/speeding-up-the-build/#makefile-parallelization-example, if it is an issue to run tests in parallel, split it into "make -j2 && make run"

crankyoldgit commented:
> maybe for make consider: https://docs.travis-ci.com/user/speeding-up-the-build/#makefile-parallelization-example, if it is an issue to run tests in parallel, split it into "make -j2 && make run"

Ooohh ... #TIL They have 2 cores. https://docs.travis-ci.com/user/reference/overview/#virtualisation-environment-vs-operating-system
I'll certainly look into this. I hope it works for platformio (but I doubt it) .. as that's where the bulk of the time is spent.

crankyoldgit committed:
Try running make in parallel.
Re: https://github.com/crankyoldgit/IRremoteESP8266/pull/1183#pullrequestreview-427077696

crankyoldgit committed:
Turn on job parallelisation for PlatformIO.
* Add `--jobs 2` to `pio run` as the travis virtual instances have 2 cores.

crankyoldgit commented:
> > maybe for make consider: https://docs.travis-ci.com/user/speeding-up-the-build/#makefile-parallelization-example, if it is an issue to run tests in parallel, split it into "make -j2 && make run"

Alas, it didn't make a difference with the platformio tasks.
But it did reduce the unit test etc task down by ~30 seconds.

crankyoldgit committed:
This time with params in the correct position.

crankyoldgit commented:
> > > maybe for make consider: https://docs.travis-ci.com/user/speeding-up-the-build/#makefile-parallelization-example, if it is an issue to run tests in parallel, split it into "make -j2 && make run"
> 
> Alas, it didn't make a difference with the platformio tasks.
> But it did reduce the unit test etc task down by ~30 seconds.

And with `pio run --jobs 2` turned on, we save about 20 seconds on one of the `pio` tasks.

Not a lot of gain, and could just be load on the travis VM host etc. But I'll leave it in.

NiKiZe commented:
pio should do it automatically from what I read, but looks good to me

NiKiZe commented:
oh... now travis don't run at all?

crankyoldgit commented:
> oh... now travis don't run at all?

Seems to be for me.
https://travis-ci.org/github/crankyoldgit/IRremoteESP8266/builds/696493991

NiKiZe commented:
Github didn't show any result from travis, so something is of

crankyoldgit commented:
> Github didn't show any result from travis, so something is of

Github does that sometimes. NFI why. :-/

NiKiZe commented:
Yepp, temporary disconnect maybe, seems to work anyhow.
(when I was trying to update the travis config the same happened, and at that time it was travis syntax erriors)
